src/components/Registration/index.jsx

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react'
+import useUser from 'hooks/useUser'
+import Input from 'templates/Form/input.jsx'
+import { capitalizeFirstLetter } from 'src/setup'
+import SubmitButton from 'templates/Form/submitButton'
+
+export default function Registration () {
+  const [email, setEmail] = useState('')
+  const [fullname, setFullnameHandler] = useState('')
+  const [password, setPassword] = useState('')
+  const [password_confirmation, setPasswordConfirmation] = useState('')
+  const [showPassword, setShowPassword] = useState(false);
+  const [showPasswordConfirmation, setShowPasswordConfirmation] = useState(false);
+  const {registrate} = useUser()
+
+  const submitHandler = (e) => {
+    e.preventDefault()
+    registrate({ email, password, password_confirmation })
+  }
+
+  const changeEmailHandler = (e) => {
+    setEmail(e.target.value)
+  }
+
+  const changePasswordHandler = (e) => {
+    setPassword(e.target.value)
+  }
+
+  const changePasswordConfirmationHandler = (e) => {
+    setPasswordConfirmation(e.target.value)
+  }
+
+  const changefullnameHandler = (e) => {
+    setFullnameHandler(e.target.value)
+  }
```

Comment:
There is no need to change the logic that you made, but just for your reference maybe a cleaner approach is to make a unique state with an object and only update the needed attribute on change.
The other clean approach is to go with a `form`.

If you go with the unique state for example:
```
const [registrationData, setRegistrationData] = useState({
email:"",
fullName:"",
password:"",
password_confirmation:""}

```
And then in your element:
```
<Input
        label={capitalizeFirstLetter(emailText)}
        type={emailText}
        value={email}
        name={emailText}
        className={emailText}
        placeholder={`Type your ${emailText}`}
        onChange={(e)=>setRegistrationData({...registrationData, email:e.target.value})}
        required={true}
      />

```